internal/lintcontext/create_contexts.go

```diff
@@ -37,21 +37,21 @@ func CreateContexts(filesOrDirs ...string) ([]LintContext, error) {
 			continue
 		}
 
-		if strings.HasSuffix(strings.ToLower(fileOrDir), ".tgz") {
-			ctx := new()
-			if err := ctx.loadObjectsFromTgzHelmChart(fileOrDir); err != nil {
-				return nil, err
-			}
-
-			contextsByDir[fileOrDir] = ctx
-			continue
-		}
-
 		err := filepath.Walk(fileOrDir, func(currentPath string, info os.FileInfo, walkErr error) error {
 			if walkErr != nil {
 				return walkErr
 			}
 			if !info.IsDir() {
+				if strings.HasSuffix(strings.ToLower(currentPath), ".tgz") {
+					ctx := new()
```

Comment:
Add deduplication to the top (since otherwise, if this same file is encountered through different arguments passed by the user, we would duplicate the work)
```go
if _, exists := contextsByDir[currentPath]; exists {
  return nil
}
```